README.md

```diff
@@ -1,25 +1,115 @@
 # glazier
 
-`glazier` is an attempt to provide a common interface to the various
-elements of different platform application frameworks. It is designed to be used
-by [druid], a UI toolkit.
+Glazier is a portable infrastructure layer intended for high quality GUI
+toolkits in Rust. It is based on druid-shell but with some changes, primarily
+around making it agnostic to the choice of drawing.
+
+Primary platforms are Windows, macOS, and Linux. Web is a secondary platform.
+Other ports may happen if they are contributed.
+
+This library is currently work in progress and should be considered experimental.
+Contributions are welcome, see [CONTRIBUTING](./CONTRIBUTING.md) for more details.
+
+## Scope
+
+The following tasks are in scope. Mostly they are implemented, but as always
+there is more refinement to be done.
+
+* Window creation, including subwindows.
```

Comment:
What is meant by "subwindows"?

Right click popups? Splash screens? Dialog popups?

Reply:
I think subwindows are a fairly common term for this. It includes context menus, combo boxes, often tooltips, and a few other similar uses. They are supported in basically all platforms (web being something of an exception).

One of the complexities here is that on modern platforms (macOS, Windows 8+, Wayland) the functionality of subwindows is to at least some extent migrating to the compositor. This is a longer term issue to deal with, and I'm explicitly marking *that* as out of scope for now.

Reply:
Wayland is a bit of a weird case compared to other platforms. Tooltips and context menu popups inherently have to work with the compositor in Wayland.

For example, a tooltip popup screenshot from firefox:
![Screenshot_20221015_163100](https://user-images.githubusercontent.com/30619168/196008324-b99be23f-4bee-4c15-8537-f3e333e32192.png)

At least for wayland these popups will be supported always, xdg-shell has a popup type the server can position relative to the parent window based on some context the client provides. The popup for windows on Wayland just acts like a special window you draw into that has some special position and input semantics.

Do other platforms treat context windows like seperate windows or is it something that has fixed possible values?

Reply:
Other platforms treat these as subwindows, generally. One reason for the subwindow/compositor distinction is that subwindows are generally supported in pre-compositor configurations (like Windows 7, for which compositor is optional), though of course with limitations such as no semi-transparency.